### PR TITLE
feat: Add a favicon, Tooltip for Avatars

### LINKED ISSUE #30: Feature: Tooltip for Avatars
### Type of feature

Feature

### Current behavior

Each card has a list of contributors with no context of who they are. I would love to add a tooltip on wh

Group 2

similar to https://github.com/open-sauced/hot/issues/21

### Desired solution

This may be accomplished through a [react tooltip](https://www.npmjs.com/package/react-tooltip) library or similar. 

1. Add a tooltip on hover
2. Ensure the [cursor-pointer](https://github.com/open-sauced/hot/blob/732680ea21028f593f3c7c35e733b5cdbece70fd/src/components/SecondaryNav.astro#L5) is added with Tailwind 

tooltip

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #36: Feature: Add a favicon
### Type of feature

Feature

### Current behavior

currently no favicon present.

Screen Shot 2021-12-23 at 3 37 13 PM

### Desired solution

Screen Shot 2021-12-23 at 3 36 57 PM

This can be grabbed from https://github.com/open-sauced/open-sauced/blob/main/public/favicon.ico

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by kelvinsekx**
## What type of PR is this? (check all applicable)

- [x] ♻️ Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description
This pr makes changes to accommodate the use of layouts best practice and add features for
1. Favicon
2. Tooltip
 

## Related Tickets & Documents
Fixes #30
Fixes #36 

## Mobile & Desktop Screenshots/Recordings
Screenshot_20211230-070027~2

Screenshot_20211230-070034~2

Favicon

Screenshot_20211230-071303~2

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [x] 🙋 no, because I need help
Nb: I will love to add test may be later
## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [ ] 📓 docs
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by mtfoley** [comment]
@kelvinsekx I deleted a bunch of the compliance comments, but typically we don't merge from the "main" branch of a fork. Since I'm seeing some merge conflicts at the moment, you may want to consider starting this PR over by keeping a local copy of your changes to refer to, and removing your fork altogether and re-creating it, so that the main history stays clean.

cc @bdougie @0-vortex

File: .eslintrc.js
Comment (bdougie) [inline .eslintrc.js:46]: What's the context here, is this for the library?

Comment (kelvinsekx) [reply .eslintrc.js:46]: eslint was complaining about windows being global here.

```bash
option === "issues" ? window.open(`${repoLink}/issues`) : window.open(repoLink);
    if (option === 'issues') return window.open(`${repoLink}/issues`);
    return window.open(repoLink);
  };
```
And setting the browser as true resolved it.

Comment (bdougie) [reply .eslintrc.js:46]: That is an opaque solution. Perhaps we could do this instead in the package.json. 

```json
"eslintConfig": {
  "globals": {
    "window": true
  }
}
```

File: package.json
Comment (bdougie) [inline package.json:58]: I don't think proptypes are providing any value in the project. The react community has moved to TypeScript, but I would also not recommend that at this time either.

File: src/components/PostList.jsx
Comment (bdougie) [inline src/components/PostList.jsx:65]: My recommendation is to remove this. There is no value-added with proptypes.

**Review by bdougie — CHANGES_REQUESTED** [review]
When reviewing the deploy preview, I noticed a few things. 

1. You cannot click the link into the GitHub user's profile. That was introduced in #29 
2. The tooltip has opacity applied. It would be preferred to make this solid since it already has the drop shadow.  
Screen Shot 2021-12-31 at 2 26 14 PM

File: package.json
Comment (kelvinsekx) [reply package.json:58]: I had to introduce propTypes since eslint screams at props whose Types are not defined.

The work around would then be to tell eslint not to scream when props are not defined? 😨

File: .eslintrc.js
Comment (kelvinsekx) [reply .eslintrc.js:46]: That's nicer.

**Comment by kelvinsekx** [comment]
> When reviewing the deploy preview, I noticed a few things.
> 
> 1. You cannot click the link into the GitHub user's profile. That was introduced in #29
> 2. The tooltip has opacity applied. It would be preferred to make this solid since it already has the drop shadow.
>    Screen Shot 2021-12-31 at 2 26 14 PM

True.

**Comment by 0-vortex** [comment]
The eslint config we are using is basically this: https://github.com/astronomer/eslint-config-astro/blob/main/index.js
I don't think these are anything more than the airbnb config, whereas open-sauced is using create-react-app preset - we should adjust the rules so it doesn't cause us trouble here, in the mean time we can work on a unified standard 🔥 
I can take that on :D

**Comment by kelvinsekx** [comment]
> The eslint config we are using is basically this: https://github.com/astronomer/eslint-config-astro/blob/main/index.js I don't think these are anything more than the airbnb config, whereas open-sauced is using create-react-app preset - we should adjust the rules so it doesn't cause us trouble here, in the mean time we can work on a unified standard 🔥 I can take that on :D

Which part are you responding too?
Is it [this](https://github.com/open-sauced/hot/pull/51#discussion_r777084936) ?

**Comment by 0-vortex** [comment]
> > The eslint config we are using is basically this: https://github.com/astronomer/eslint-config-astro/blob/main/index.js I don't think these are anything more than the airbnb config, whereas open-sauced is using create-react-app preset - we should adjust the rules so it doesn't cause us trouble here, in the mean time we can work on a unified standard 🔥 I can take that on :D
> 
> Which part are you responding too? Is it [this](https://github.com/open-sauced/hot/pull/51#discussion_r777084936) ?

Exactly. Astro is under heavy development as well, it's good to stay flexible 🍕

**Comment by kelvinsekx** [comment]
> > > The eslint config we are using is basically this: https://github.com/astronomer/eslint-config-astro/blob/main/index.js I don't think these are anything more than the airbnb config, whereas open-sauced is using create-react-app preset - we should adjust the rules so it doesn't cause us trouble here, in the mean time we can work on a unified standard 🔥 I can take that on :D
> > 
> > 
> > Which part are you responding too? Is it [this](https://github.com/open-sauced/hot/pull/51#discussion_r777084936) ?
> 
> Exactly. Astro is under heavy development as well, it's good to stay flexible 🍕

Would you rather make a self-assigned issue for that?
Or I should just 'eslint-ignore" the files, make corrections and send back a PR?

**Comment by kelvinsekx** [comment]
```bash
npm ERR! code ETARGET
npm ERR! notarget No matching version found for astro@^0.22.1.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
```

@0-vortex Please I need an help on this. I keep getting this error after i tried manually merging

**Comment by 0-vortex** [comment]
> ```shell
> ```shell
> npm ERR! code ETARGET
> npm ERR! notarget No matching version found for astro@^0.22.1.
> npm ERR! notarget In most cases you or one of your dependencies are requesting
> npm ERR! notarget a package version that doesn't exist.
> ```
> 
> 
>     
>       
>     
> 
>       
>     
> 
>     
>   
> @0-vortex Please I need an help on this. I keep getting this error after i tried manually merging
> ```

Please use the `npm-shrinkwrap` from master and do a `npm ci` to get the new dependencies, that should fix it!

**Comment by kelvinsekx** [comment]
```bash
npm ERR! code ETARGET
npm ERR! notarget No matching version found for astro@^0.22.1.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
```

@0-vortex Please I need an help on this. I keep getting this error after i tried manually merging

> > ```shell
> > ```shell
> > npm ERR! code ETARGET
> > npm ERR! notarget No matching version found for astro@^0.22.1.
> > npm ERR! notarget In most cases you or one of your dependencies are requesting
> > npm ERR! notarget a package version that doesn't exist.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > @0-vortex Please I need an help on this. I keep getting this error after i tried manually merging
> 
> Please use the `npm-shrinkwrap` from master and do a `npm ci` to get the new dependencies, that should fix it! After that re-install the npm packages your PR is adding via `npm install --save`
@0-vortex after doing as instructed, I resolved the error. But encountered another one
```bash
'astro' is not recognized as an internal or external command,
operable program or batch file.
```

using `npm list --dev`, the packages is listed. Quite confusing

**Comment by 0-vortex** [comment]
are you using node>14 and npm>8 ?

**Comment by mtfoley** [comment]
@kelvinsekx it looks like your `npm-shrinkwrap.json` has merge conflict annotations in it.

I think others might do it differently, but I'd suggest this:
```bash
# download and overwrite npm-shrinkwrap from main
wget -O npm-shrinkwrap.json https://github.com/open-sauced/open-sauced/raw/main/npm-shrinkwrap.json
# rerun npm ci to pickup packages
npm ci
npm test
npm start
```

**Comment by kelvinsekx** [comment]
> are you using node>14 and npm>8 ?

v16.13.0

**Comment by 0-vortex** [comment]
> @kelvinsekx it looks like your `npm-shrinkwrap.json` has merge conflict annotations in it.
> 
> 
> 
> I think others might do it differently, but I'd suggest this:
> 
> ```bash
> 
> # download and overwrite npm-shrinkwrap from main
> 
> wget -O npm-shrinkwrap.json https://github.com/open-sauced/open-sauced/raw/main/npm-shrinkwrap.json
> 
> # rerun npm ci to pickup packages
> 
> npm ci
> 
> npm test
> 
> npm start
> 
> ```
> 
> 

replace open-sauced/open-sauced with open-sauced/hot 🤓

**Comment by kelvinsekx** [comment]
> > @kelvinsekx it looks like your `npm-shrinkwrap.json` has merge conflict annotations in it.
> > I think others might do it differently, but I'd suggest this:
> > ```shell
> > # download and overwrite npm-shrinkwrap from main
> > 
> > wget -O npm-shrinkwrap.json https://github.com/open-sauced/open-sauced/raw/main/npm-shrinkwrap.json
> > 
> > # rerun npm ci to pickup packages
> > 
> > npm ci
> > 
> > npm test
> > 
> > npm start
> > ```
> 
> replace open-sauced/open-sauced with open-sauced/hot 🤓

Thanks I will give that a try... 😂

**Comment by 0-vortex** [comment]
If you have the repository set up with `upstream` remote you can do:

```shell
git restore package.json npm-shrinkwrap.json -s upstream/main
npm install --save react-tooltip prop-types
```

Alternatively, update your fork's `main` branch and do:

```shell
git restore package.json npm-shrinkwrap.json -s origin/main
npm install --save react-tooltip prop-types
```

Essentially what you want to do is merge with upstream, keeping those changes and adding your features on top of it 🍕

File: .eslintrc.js
Comment (0-vortex) [reply .eslintrc.js:46]: put this in `.eslintrc.js` please!

**Pull request closed without merging**
